test/Common.cpp

```diff
@@ -91,22 +91,27 @@ CommonOptions::CommonOptions(std::string _caption):
 		po::options_description::m_default_line_length,
 		po::options_description::m_default_line_length - 23
 	)
+{
+
+}
+
+void CommonOptions::addOptions()
 {
 	options.add_options()
 		("evm-version", po::value(&evmVersionString), "which evm version to use")
 		("testpath", po::value<fs::path>(&this->testPath)->default_value(solidity::test::testPath()), "path to test files")
 		("vm", po::value<std::vector<fs::path>>(&vmPaths), "path to evmc library, can be supplied multiple times.")
-		("ewasm", po::bool_switch(&ewasm), "tries to automatically find an ewasm vm and enable ewasm test-execution.")
-		("no-semantic-tests", po::bool_switch(&disableSemanticTests), "disable semantic tests")
-		("no-smt", po::bool_switch(&disableSMT), "disable SMT checker")
-		("optimize", po::bool_switch(&optimize), "enables optimization")
-		("enforce-via-yul", po::bool_switch(&enforceViaYul), "Enforce compiling all tests via yul to see if additional tests can be activated.")
-		("enforce-compile-to-ewasm", po::bool_switch(&enforceCompileToEwasm), "Enforce compiling all tests to Ewasm to see if additional tests can be activated.")
-		("enforce-gas-cost", po::bool_switch(&enforceGasTest), "Enforce checking gas cost in semantic tests.")
-		("enforce-gas-cost-min-value", po::value(&enforceGasTestMinValue), "Threshold value to enforce adding gas checks to a test.")
-		("abiencoderv1", po::bool_switch(&useABIEncoderV1), "enables abi encoder v1")
-		("show-messages", po::bool_switch(&showMessages), "enables message output")
-		("show-metadata", po::bool_switch(&showMetadata), "enables metadata output");
+		("ewasm", po::bool_switch(&ewasm)->default_value(ewasm), "tries to automatically find an ewasm vm and enable ewasm test-execution.")
+		("no-semantic-tests", po::bool_switch(&disableSemanticTests)->default_value(disableSemanticTests), "disable semantic tests")
+		("no-smt", po::bool_switch(&disableSMT)->default_value(disableSMT), "disable SMT checker")
+		("optimize", po::bool_switch(&optimize)->default_value(optimize), "enables optimization")
+		("enforce-via-yul", po::bool_switch(&enforceViaYul)->default_value(enforceViaYul), "Enforce compiling all tests via yul to see if additional tests can be activated.")
+		("enforce-compile-to-ewasm", po::bool_switch(&enforceCompileToEwasm)->default_value(enforceCompileToEwasm), "Enforce compiling all tests to Ewasm to see if additional tests can be activated.")
+		("enforce-gas-cost", po::bool_switch(&enforceGasTest)->default_value(enforceGasTest), "Enforce checking gas cost in semantic tests.")
+		("enforce-gas-cost-min-value", po::value(&enforceGasTestMinValue)->default_value(enforceGasTestMinValue), "Threshold value to enforce adding gas checks to a test.")
```

Comment:
So without `default_value()` `program_options` just overwrites the original value of the variable? Does this mean that the default cost was being ignored so far?

Reply:
The default cost was previously being used because it was being directly assigned to at the end of the parse chain, it was actually user-defined cost that was being ignored. This change allows default values to be determined at construction time and then propagated to the `options_description` object via the `default_value` calls so `program_options` controls all defaulting behavior and still accepts user input instead of direct assignment overwriting those values post-parse.

Reply:
> The default cost was previously being used because it was being directly assigned to at the end of the parse chain, it was actually user-defined cost that was being ignored.

That's only for `isoltest`. `soltest` was not overwriting it.

The change is fine. Even if it does not change behavior, it makes it more explicit, which is good. I'm just trying to figure out if the previous code was actually taking the defaults into account or just overwriting them.

Reply:
Yes, `soltest` was always functioning as expected because the defaults were implicit via the default values the struct was initialized with, `default_value` just makes it explicit.